codalab/apps/web/static/js/search/ws_actions.js

```diff
@@ -98,13 +98,18 @@ var WorksheetActions =  function() {
                                 if (data.data.stdout){
                                     console.log(data.data.stdout);
                                     term.echo(data.data.stdout);
-                                    var references = data.data.structured_result['refs'];
-                                    Object.keys(references).forEach(function(k) {
-                                        $(".terminal-output div div:contains(" + k + ")").html(function(idx, html) {
-                                            var link = '/' + references[k]['type'] + 's/' + references[k]['uuid'];
-                                            return html.replace(k, "<a href=" + link + " target='_blank'>" + k + "</a>");
-                                        });
-                                    }, this);
+                                    if (data.data.structured_result && data.data.structured_result.refs){
+                                        var references = data.data.structured_result['refs'];
+                                        Object.keys(references).forEach(function(k) {
+                                            $(".terminal-output div div:contains(" + k + ")").html(function(idx, html) {
+                                                var hyperlink_info = references[k];
+                                                if (hyperlink_info.uuid && (hyperlink_info.type === 'bundle' || hyperlink_info.type === 'worksheet')) {
+                                                    var link = '/' + hyperlink_info['type'] + 's/' + hyperlink_info['uuid'];
+                                                    return html.replace(k, "<a href=" + link + " target='_blank'>" + k + "</a>");
+                                                }
```

Comment:
If there's an error, don't just fail silently!  Log it to some place and also have selenium check this.  Also, remember to fail gracefully (this is not happening now).